fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/operations/CreateNonRdfSourceOperationBuilder.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.api.operations;
+
+import java.net.URI;
+import java.util.Collection;
+
+/**
+ * Builder for operations to create non-rdf sources
+ *
+ * @author bbpennel
+ */
+public interface CreateNonRdfSourceOperationBuilder extends NonRdfSourceOperationBuilder {
+
+    @Override
+    CreateNonRdfSourceOperationBuilder mimeType(String mimetype);
+
+    @Override
+    CreateNonRdfSourceOperationBuilder filename(String filename);
+
+    @Override
+    CreateNonRdfSourceOperationBuilder contentDigests(Collection<URI> digests);
+
+    @Override
+    CreateNonRdfSourceOperationBuilder contentSize(Long size);
+
+    /**
+     * Set the parent identifier of the resource
+     *
+     * @param parentId parent internal identifier
+     * @return the builder
+     */
+    CreateNonRdfSourceOperationBuilder parentId(String parentId);
```

Comment:
Feels like this could move up to `ResourceOperationBuilder` as you define it here and in the `CreateRdfSourceOperationBuilder`. Parent is general enough it fits both.

Reply:
I had it here since we are no longer allowing resources to be moved after they are created, so only Create operations need to be able to pass a parent id down. As we talked about a little on slack, an alternative would be to make the builders less restrictive, allowing creation related fields to be set on update operations, etc, and relying either on the code calling it or after it make good decisions. Definitely open to feedback on this

Reply:
I'm in favor of erring on the side of too strict and then loosen it up if there is a need.